### DISCUSSION TITLE
Disable auto-sourcing directories?

### DISCUSSION THREAD

**Discussion by jason5122** [scop/bash-completion#745] [Q&A]
Hi, I'm experiencing massive slowdowns from sourcing every completion file in `/usr/local/etc/bash_profile.d` on macOS — just to be clear, this is no fault of `bash-completion`. These files are added to this directory automatically after installing the relevant programs in `brew`, so manually removing them is a bit of a hassle.

Is there a proper way to disable this behavior? Right now, I'm doing `export BASH_COMPLETION_COMPAT_DIR=' '` (a single space), which seems a bit hacky.

**Comment by scop** [comment]
Setting `BASH_COMPLETION_COMPAT_DIR` to a nonexistent or to a known empty dir is the way to accomplish that at the moment. The next version will have a [`BASH_COMPLETION_COMPAT_IGNORE` setting](https://github.com/scop/bash-completion/blob/966a4e043822613040546e8c003509798c5fae1a/doc/configuration.md#bash_completion_compat_ignore) which can additionally be used for this and more selective exclusion.

**Reply by scop** [reply]
I don't have an affected system at hand, but I guess at least some of those completion files added by brew packages only work with bash-completion 2+. Those files would be better off installed in the non-compat completion dir where the snippets supported by us are in (`pkg-config --variable=completionsdir bash-completion`) so they'd be only loaded on demand. Having someone go those through and file bugs for respective things in brew asking them to be moved would be great. (Files intended to work with 1.x too need to be installed in the location you mention.)
